chef/cookbooks/neutron/recipes/common_config.rb

```diff
@@ -120,7 +120,10 @@
   service_plugins = "#{service_plugins}, neutron.services.loadbalancer.plugin.LoadBalancerPlugin"
 end
 
-network_nodes = search_env_filtered(:node, "roles:neutron-network")
+network_nodes_count = 1
+if neutron[:neutron][:elements_expanded]
+  network_nodes_count = neutron[:neutron][:elements_expanded]["neutron-network"].count
+end
 
```

Comment:
This was my first version but the line is extremely long. So I prefer this syntax.